R/external-deps.R

```diff
@@ -53,21 +53,41 @@ external_dependencies <- function() {
         )
       )
     ),
+    math_dependency(pkg, call = call)
+  ))
+}
+
+math_dependency <- function(pkg, call = caller_env()) {
+  math <- config_math_rendering(pkg)
+  if (math == "mathjax") {
     cached_dependency(
       name = "MathJax",
-      version = "2.7.5",
+      version = "3.2.2",
+      files = list(
+        list(
+          url = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/3.2.2/es5/tex-chtml.min.js",
```

Comment:
If we switch to using latest, we'll need to fundamentally rethink how caching works, which I don't think is worth it (for now).

Reply:
Ideally, we would get the file content's hash from some `HEAD` request's header value. But there's no such header included in the response, `curl -I https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js` gives me:

```sh
HTTP/2 200 
access-control-allow-origin: *
access-control-expose-headers: *
timing-allow-origin: *
cache-control: public, max-age=604800, s-maxage=43200
cross-origin-resource-policy: cross-origin
x-content-type-options: nosniff
strict-transport-security: max-age=31536000; includeSubDomains; preload
content-type: application/javascript; charset=utf-8
x-jsd-version: 3.2.2
x-jsd-version-type: version
etag: W/"11e60f-0zusawQQZK5DMNzC2Vjr5MKOvlg"
accept-ranges: bytes
age: 26759
date: Wed, 29 May 2024 14:31:56 GMT
x-served-by: cache-fra-etou8220064-FRA, cache-ams21031-AMS
x-cache: HIT, HIT
vary: Accept-Encoding
alt-svc: h3=":443";ma=86400,h3-29=":443";ma=86400,h3-27=":443";ma=86400
content-length: 1173007
```

Instead, we could do cache invalidation based on the [ETag header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag) value.

Or we could leverage the `x-jsd-version` header (I'd actually prefer this over ETag). Apparently, it's [jsDelivr](https://github.com/jsdelivr/jsdelivr)-specific and gives the asset version (though I couldn't really find any documentation apart from [this comment](https://github.com/jsdelivr/jsdelivr/issues/18376#issuecomment-1047040896) mentioning it).

I think it wouldn't require too much refactoring of the current code. I could give it a shot if you like.

Reply:
Yes, we could switch to HTTP caching, but I think we can do that after this PR.